src/lib/utils/application-commands/ApplicationCommandRegistries.ts

```diff
@@ -53,8 +58,119 @@ export async function handleRegistryAPICalls() {
 		}
 	}
 
-	const { applicationCommands, globalCommands, guildCommands } = await getNeededRegistryParameters(allGuildIdsToFetchCommandsFor);
+	if (defaultBehaviorWhenNotIdentical === RegisterBehavior.BulkOverwrite) {
+		await handleBulkOverwrite(commandStore, container.client.application!.commands);
```

Comment:
```suggestion
		const applicationCommandManager = container.client.application?.commands;
		if (applicationCommandManager) {
			await handleBulkOverwrite(commandStore, applicationCommandManager);
		} else {
			bulkOverwriteFatal('Failed to get client application for bulk overwriting! Was your client successfully logged in?');
		}
```

And create and import method `bulkOverwriteFatal`

Reply:
This is a non issue as the method can only be called by us post ready

Reply:
Runtime checks are still better than bypassing them at compile them imo

Reply:
And yet a lot of code doesn't do that... I don't disagree with you on the concept, but given this internal method I'm leaning towards leaving it as is (also outside scope of PR as then I'd have to also do it to the other behaviors)